tests/nexus.tests.ps1

```diff
@@ -0,0 +1,248 @@
+            $expectedConfiguation = @'
+# Jetty section
+# application-port=8081
+# application-host=0.0.0.0
+# nexus-args=${jetty.etc}/jetty.xml,${jetty.etc}/jetty-http.xml,${jetty.etc}/jetty-requestlog.xml
+# nexus-context-path=/
+
+# Nexus section
+# nexus-edition=nexus-pro-edition
+# nexus-features=\
+#  nexus-pro-feature
+
+# nexus.hazelcast.discovery.isEnabled=true
+jetty.https.stsMaxAge=-1
+application-port-ssl=8443
+nexus-args=${jetty.etc}/jetty.xml,${jetty.etc}/jetty-https.xml,${jetty.etc}/jetty-requestlog.xml
+
+'@
+        }
+    
+        It "Service has HSTS disabled" {
+            $ConfigurationFile[12] -eq 'jetty.https.stsMaxAge=-1' | Should -Be $true
+        }
+
+        It "Service is using port 8443" {
+            $ConfigurationFile[13] -eq 'application-port-ssl=8443' | Should -Be $true
+        }
+
+        It "Service is using jetty-https.xml" {
+            $ConfigurationFile[14] -eq 'nexus-args=${jetty.etc}/jetty.xml,${jetty.etc}/jetty-https.xml,${jetty.etc}/jetty-requestlog.xml' | Should -Be $true
+        }
+
+        It "Service is using the appropriate SSL Certificate" {
+            $certStoreCertificate -eq $serviceCertificate | Should -Be $true
+        }
+
+        It "Service responds to web requests" {
+            ([System.Net.WebRequest]::Create("https://$($Fqdn):8443") -as [System.net.HttpWebRequest]).GetResponse().StatusCode -eq 'OK' | Should -Be $true
+        }
+    }
+
+    Context "Repository Configuration" {
+        BeforeAll {
+            $password = (Get-Content 'C:\ProgramData\sonatype-work\nexus3\admin.password') | ConvertTo-SecureString -AsPlainText -Force
+            $credential = [System.Management.Automation.PSCredential]::new('admin',$password)
+            . "C:\choco-setup\files\scripts\Get-Helpers.ps1"
+            $null = Connect-NexusServer -Hostname $Fqdn -Credential $credential -UseSSL
+
+            $repositories = Get-NexusRepository
+        }
+        It "ChocolateyInternal" {
+            'ChocolateyInternal' -in $repositories.Name | Should -Be $true
+        }
+
+        It "ChocolateyTest" {
+            'ChocolateyTest' -in $repositories.Name | Should -Be $true
+        }
+
+        It "choco-install" {
+            'choco-install' -in $repositories.Name | Should -Be $true
+        }
+
+        It 'NuGet API-Key Realm is active' {
+            'NuGetApiKey' -in (Get-NexusRealm -Active).id | Should -Be $true
+        }
+    }
+
+    Context "Package Availability" {
+        BeforeAll {
+            $packages = choco list -s ChocolateyInternal -r | ConvertFrom-Csv -Delimiter '|' -Header Package,Version
+        }
+
+        It "Chocolatey is in the repository" {
+            'chocolatey' -in $packages.Package | Should -Be $true
+        }
+
+        It "Chocolatey is version 1.1.0" {
+            ($packages | Where-Object Package -eq 'chocolatey').Version | Should -Be '1.1.0'
```

Comment:
For these package version tests. Would it be a better idea to write these as should be greater than or equal to the version number specified?

Reply:
This could be complicated... but it's a very good point... we'll want to either pin the version, or make sure we're testing the version properly. You could look at the `common-helpers` used in the choco tests for how to check version...